crates/rmf_site_editor/src/site/property_plugin.rs

```diff
@@ -200,3 +326,93 @@ fn on_add_root_scenario<T: Property, F: QueryFilter + 'static + Send + Sync>(
 
     T::insert_on_new_scenario(trigger.target(), world);
 }
+
+/// Handles cleanup of scenario modifiers when elements are despawned
+pub fn on_remove_element<E: Element>(
+    trigger: Trigger<OnDespawn, E>,
+    scenarios: Query<Entity, With<ScenarioModifiers<Entity>>>,
+    mut commands: Commands,
+) {
+    for scenario_entity in scenarios.iter() {
+        commands.trigger(RemoveModifier::new(trigger.target(), scenario_entity));
+    }
+}
+
+/// Whenever a Modifier<T> component has been removed from a modifier entit`y, this
+/// observer checks that the entity is not empty (has no other Modiifers). If
+/// empty, the modifier entity will be sent for removal and despawn.
+fn on_remove_modifier<T: Property>(
```

Comment:
Is there a particular need to remove an empty modifier entity? If we just leave it in place, then it may get reused later rather than needing to respawn the modifier.

Reply:
Yeah I think we can leave empty modifier entities alone. I would still remove modifier entities if the affiliated instances (or elements in general) are deleted from the site.

[2ba7ea3](https://github.com/open-rmf/rmf_site/pull/325/commits/2ba7ea3538df3eea80f24e4e15e9192ce7d30ec1)